File: src/main/java/seremis/geninfusion/soul/gene/newAI/wander/GeneAIWander.java
package seremis.geninfusion.soul.gene.newAI.wander;

import seremis.geninfusion.api.soul.IAllele;
import seremis.geninfusion.api.soul.lib.Genes;
import seremis.geninfusion.soul.MasterGene;
import seremis.geninfusion.soul.allele.AlleleBoolean;

public class GeneAIWander extends MasterGene {

    public GeneAIWander() {
        addControlledGene(Genes.GENE_AI_WANDER_INDEX);
        addControlledGene(Genes.GENE_AI_WANDER_MOVE_SPEED);
    }

    @Override
    public Class<? extends IAllele> possibleAlleles() {
        return AlleleBoolean.class;
    }
}

File: src/main/scala/seremis/geninfusion/soul/gene/newAI/GeneAIWander.scala
package seremis.geninfusion.soul.gene.newAI

import seremis.geninfusion.api.lib.Genes
import seremis.geninfusion.soul.MasterGene

class GeneAIWander extends MasterGene(classOf[Boolean]) {
    addControlledGene(Genes.GeneAIWanderIndex)
    addControlledGene(Genes.GeneAIWanderMoveSpeed)
}
